src/Identity/IdentityServer/BaseRequestValidator.cs

```diff
@@ -87,6 +91,11 @@ public abstract class BaseRequestValidator<T> where T : class
         _tokenDataFactory = tokenDataFactory;
         FeatureService = featureService;
         SsoConfigRepository = ssoConfigRepository;
+        _distributedCache = distributedCache;
+        _cacheEntryOptions = new DistributedCacheEntryOptions
+        {
+            AbsoluteExpirationRelativeToNow = new TimeSpan(0, 15, 0)
```

Comment:
❓ Is this the TOTP expiration window? Centralizing this setting or using whatever is configured elsewhere would help.

Reply:
The issue sloved here is Authenticator TOTPs have a 1 minute TTL, and email TOTPs has a 5 minute TTL. Based on the RFC the recomendation is to accept TOTPs where T = T+1, where T is the vaild time-window step. Meaning that all TOTPs are vaild for the window they are created in and the next (T+1). This means an Authenticator TOTP will be vaild for 2 minutes, and an email TOTP will be vaild for 10 minutes. This is how it is implemented in OTP.NET library we use.

Chaching the TOTPs for 15 minutes ensures an Email and Authenticator TOTP is well past the T+1 mark. It also ensures if the clocks are misaligned between servers the cache will not clear too soon and allow the use of a TOTP twice.

The TTL for the cache being hardcoded gives code smells but I believe in this situation it's permissible.

Reply:
Great explanation, thanks. I think a brief comment in code could help here.

Reply:
Comment added.